app/controllers/application_controller/tags.rb

```diff
@@ -222,42 +194,39 @@ def tag_edit_build_screen
       end
     end
 
-    # Set to first category, if not already set
-    @edit[:cat] ||= cats.first
-
-    unless @object_ids.blank?
-      @tagitems = @tagging.constantize.where(:id => @object_ids).sort_by { |t| t.name.try(:downcase).to_s }
-    end
+    @tagitems = @tagging.constantize.where(:id => @object_ids).sort_by { |t| t.name.try(:downcase).to_s }
 
     @view = get_db_view(@tagging, :clickable => false) # Instantiate the MIQ Report view object
     @view.table = ReportFormatter::Converter.records2table(@tagitems, @view.cols + ['id'])
 
-    # Start with the first items assignments
-    @edit[:new][:assignments] =
-      Classification.find_assigned_entries(@tagitems[0]).collect { |e| e.id unless e.parent.read_only? }
-    @tagitems.each do |item|
-      itemassign = Classification.find_assigned_entries(item).collect(&:id) # Get each items assignments
-      @edit[:new][:assignments].delete_if { |a| !itemassign.include?(a) } # Remove any assignments that are not in the new items assignments
-      break if @edit[:new][:assignments].length == 0                      # Stop looking if no assignments are left
+    @edit[:new][:assignments] = @assignments = @tagitems.map do |tagitem|
```

Comment:
Do you really need the `@assignments` here if you store it in the session as well?